src/OrbitGl/MultivariateTimeSeries.cpp

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2023 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "OrbitGl/MultivariateTimeSeries.h"
+
+#include "OrbitBase/Logging.h"
+
+namespace orbit_gl {
+
+MultivariateTimeSeries::MultivariateTimeSeries(absl::Span<const std::string> series_names,
+                                               uint8_t value_decimal_digits, std::string value_unit)
+    : series_names_(series_names.begin(), series_names.end()),
+      value_decimal_digits_{value_decimal_digits},
+      value_unit_{std::move(value_unit)} {
+  ORBIT_CHECK(!series_names.empty());
+}
+
+double MultivariateTimeSeries::GetMin() const {
+  absl::MutexLock lock(&mutex_);
+  return min_;
+}
+
+double MultivariateTimeSeries::GetMax() const {
+  absl::MutexLock lock(&mutex_);
+  return max_;
+}
+
+bool MultivariateTimeSeries::IsEmpty() const {
+  absl::MutexLock lock(&mutex_);
+  return time_to_series_values_.empty();
+}
+
+size_t MultivariateTimeSeries::GetTimeToSeriesValuesSize() const {
+  absl::MutexLock lock(&mutex_);
+  return time_to_series_values_.size();
+}
+
+uint64_t MultivariateTimeSeries::StartTimeInNs() const {
+  absl::MutexLock lock(&mutex_);
+  ORBIT_CHECK(!time_to_series_values_.empty());
+  return time_to_series_values_.begin()->first;
+}
+
+uint64_t MultivariateTimeSeries::EndTimeInNs() const {
+  absl::MutexLock lock(&mutex_);
+  ORBIT_CHECK(!time_to_series_values_.empty());
+  return time_to_series_values_.rbegin()->first;
+}
+
+const std::vector<double>& MultivariateTimeSeries::GetPreviousOrFirstEntry(uint64_t time) const {
+  absl::MutexLock lock(&mutex_);
+  return GetPreviousOrFirstEntryIterator(time)->second;
```

Comment:
You're returning a reference to some mutex-protected data. This doesn't make a lot of sense, because now the caller has a pointer to the mutex-protected data, but doesn't know that the data should be mutex protected.

Could you instead return a copy? Or is this performance relevant?